onnxruntime/test/providers/provider_test_utils.h

```diff
@@ -539,18 +572,33 @@ class OpTester {
   void AddData(std::vector<Data>& data, const char* name, const std::vector<int64_t>& dims, const T* values,
                int64_t values_count, bool is_initializer = false, bool sort_output = false,
                const std::vector<std::string>* dim_params = nullptr,
-               float rel_error = 0.0f, float abs_error = 0.0f) {
+               float rel_error = 0.0f, float abs_error = 0.0f, bool is_optional_type_tensor = false) {
     ORT_TRY {
       TensorShape shape{dims};
-      ORT_ENFORCE(shape.Size() == values_count, values_count, " input values doesn't match tensor size of ",
+
+      // Value being null means we are creating an optional type tensor of None
+      ORT_ENFORCE(values == nullptr ||
+                      shape.Size() == values_count,
```

Comment:
should it check is_optional_type_tensor if allowing nullptr to be valid?

Reply:
Good point, did that, thanks

Reply:
EDIT: I think this is not possible. Even in case of non-optional type data, values may be nullptr when the input/output is empty (`TokenizerCharLevel_EmptyOutputC`). I refined the check a bit so as to ensure that when values is nullptr and the type is a non-optional type, `shape.size()` should still equal `values_count`. (I presume that was the concern that prompted this comment)